spectrochempy/analysis/kinetic_utilities.py

```diff
@@ -44,14 +61,15 @@ class SolverError(SpectroChemPyError):
 # Utility
 # --------
 def _interpret_equation(eq, species):
-    # transform an equation given as a string to a dictionary of species with
-    # integer stoechiometric coefficients.
+    # transform an equation given as a string to dictionaries of species with
+    # integer stoechiometric coefficients for the left (reactants) and right (products)
+    # side of the equation.
 
     regex = r"(((([\.,0-9]*)((?=[a-zA-Z])[a-zA-Z, 1-9]+))(?=\+?))?(?=(->|→)?))"
 
     matches = re.finditer(regex, eq.replace(" ", ""))
-    equation = {}
-    mult = -1  # reactants
+    left, right = {}, {}
```

Comment:
A and B matrices (left and right stoichiometries) must be determined separately, else reactions such as A + B = 2 A + C are not handled properly